src/controllers/dynakube/istio/client.go

```diff
@@ -0,0 +1,180 @@
+package istio
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/pkg/errors"
+	istiov1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	istioclientset "istio.io/client-go/pkg/clientset/versioned"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type ClientBuilder func(config *rest.Config, scheme *runtime.Scheme, namespace string) (*Client, error)
+
+// Client - an adapter for the external istioclientset library
+type Client struct {
+	IstioClient istioclientset.Interface
+	Scheme      *runtime.Scheme
+	Namespace   string
+}
+
+func NewClient(config *rest.Config, scheme *runtime.Scheme, namespace string) (*Client, error) {
+	istioClient, err := istioclientset.NewForConfig(config)
+
+	if err != nil {
+		log.Info("failed to initialize istio client", "error", err.Error())
+		return nil, errors.WithStack(err)
+	}
+	return &Client{
+		IstioClient: istioClient,
+		Scheme:      scheme,
+		Namespace:   namespace,
+	}, nil
+}
+
+var _ ClientBuilder = NewClient
+
+// TODO: Maybe move whole check here
+func (cl *Client) Discovery() discovery.DiscoveryInterface {
+	return cl.IstioClient.Discovery()
+}
+
+func (cl *Client) GetVirtualService(ctx context.Context, name string) (*istiov1alpha3.VirtualService, error) {
+	virtualService, err := cl.IstioClient.NetworkingV1alpha3().VirtualServices(cl.Namespace).Get(ctx, name, metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
+		log.Info("failed to get current virtual service", "name", name, "error", err.Error())
+		return nil, errors.WithStack(err)
+	}
+	return virtualService, nil
+}
+
+func (cl *Client) ApplyVirtualService(ctx context.Context, owner metav1.Object, newVirtualService *istiov1alpha3.VirtualService) error {
```

Comment:
```suggestion
func (cl *Client) CreateVirtualService(ctx context.Context, owner metav1.Object, newVirtualService *istiov1alpha3.VirtualService) error {
```

Reply:
jsut to respect other naming convention: Get/Create/Delete/Update (CRUD like ;)

Reply:
"create" is bit misleading for me, as it also updates it if its already there
also "apply" is also not 100% true, because that would mean I use a "patch" which I don't 😅 

will go with `CreateOrUpdate...` as we already use that naming, and I can't come up with anything better

Reply:
done